pkg/networkservice/core/next/context.go

```diff
@@ -47,6 +51,9 @@ func Server(ctx context.Context) networkservice.NetworkServiceServer {
 	if rv, ok := ctx.Value(nextServerKey).(networkservice.NetworkServiceServer); ok {
 		return rv
 	}
+	if rv, ok := ctx.Value(nextClientKey).(networkservice.NetworkServiceClient); ok {
```

Comment:
So I'm curious, why are we converting nextClient to nextServer here?  Perhaps I still need my morning coffee... but I've lost the thread :)

Reply:
It is needed for the case when we have for example a chain of servers and one adpated client
```
s := next.NewNetworkServiceServer(
something.NewServer(), 
adapters.NewClientToServer(something.NewClient()),
something.NewServer())
```
And we want that this adapted client can correctly call the next server.

Note: This client in `Request()` and `Close()`  uses  `next.Client(ctx)`. So if we will not adapt here we will break a chain.

Reply:
Excellent!  Thank you for the patient explanation :)